ProcessOut/Classes/ProcessOut.swift

```diff
@@ -358,6 +358,33 @@ public class ProcessOut {
             makeCardPayment(invoiceId: invoiceId, token: newSource, handler: handler, with: with)
         })
     }
+
+    /// Initiate a payment authorization from a previously generated invoice and card token
+    ///
+    /// - Parameters:
+    ///   - AuthorizationRequest: contains all the necessary fields to initiate an authorisation request.
+    ///   - handler: Custom 3DS2 handler (please refer to our documentation for this)
+    ///   - with: UIViewController to display webviews and perform fingerprinting
+    public static func makeCardPayment(AuthorizationRequest: AuthorizationRequest, handler: ThreeDSHandler, with: UIViewController) {
+        // Handle empty source
+        if (AuthorizationRequest.getSource().isEmpty()) {
+            handler.onError(error: ProcessOutException.InternalError)
+            return
+        }
+
+        guard let body = try? JSONEncoder().encode(AuthorizationRequest) else {
+            handler.onError(error: ProcessOutException.InternalError)
+            return
+        }
+
+        guard let json = try? JSONSerialization.jsonObject(with: body, options: []) as? [String : Any] else {
+            handler.onError(error: ProcessOutException.InternalError)
+            return
+        }
+        makeAuthorizationRequest(invoiceId: invoiceId, thirdPartySDKVersion: thirdPartySDKVersion, json: json, handler: handler, with: with, actionHandlerCompletion: { (newSource) in
```

Comment:
Have to change this line to `makeAuthorizationRequest(AuthorizationRequest: AuthorizationRequest....`

Also overload the `makeAuthorizationRequest` function to accept `AuthorizationRequest: AuthorizationRequest` believe it might cause issues